### ISSUE TITLE
Deleting runners results in a stuck terminated state

### ISSUE THREAD

**Issue by aroemen** [evryfs/github-actions-runner-operator#212]
Running `kubectl apply -f .\gh-runners-linux.yaml` creates the runners as expected in my GitHub organization. When I delete them though (using `kubectl delete -f .\gh-runners-linux.yaml`), the pods that contained the runners get stuck in a "Terminating" status.

```
NAMESPACE                        NAME                                              READY   STATUS        RESTARTS   AGE
github-action-runners            runner-pool-pod-fhthp                             0/3     Terminating   0          4m50s
github-action-runners            runner-pool-pod-wfs62                             0/3     Terminating   0          4m50s
github-actions-runner-operator   github-actions-runner-operator-59b9d486b5-t2p62   1/1     Running       0          5m26s
```

If I edit the pod and remove the finalizer (garo.tietoevry.com/runner-registration), the pod successfully deletes after saving that change. The runner is not being removed from my list of GitHub self hosted runners though as I would expect. Am I missing something here?

**Comment by davidkarlsen** [comment]
I was able to reproduce it. It's an edge case when you delete the actual cr. In this case it's gone and the cleanup step handling the finalization https://github.com/evryfs/github-actions-runner-operator/blob/master/controllers/githubactionrunner_controller.go#L116 is not reached.

**Comment by aroemen** [comment]
What would be another way to tear down these resources then?

**Comment by duyhenryer** [comment]
Hi there, 
I have the same issue here
```
NAME                    READY   STATUS        RESTARTS   AGE
runner-pool-pod-7qhqc   0/3     Terminating   0          4d6h
runner-pool-pod-d96bw   0/3     Terminating   0          4h38m
runner-pool-pod-w278v   0/3     Terminating   0          4h38m
runner-pool-pod-xbmww   0/3     Terminating   0          4h47m
```
I can't remove them.
Thank you.

**Comment by davidkarlsen** [comment]
yes, and that's what the operator does after de-registering them from github - which is why I am curious what the operator logs.

**Comment by aroemen** [comment]
@davidkarlsen I posted the operator logs back in March. Do you need additional data?

**Comment by davidkarlsen** [comment]
@aroemen sorry, commented on the wrong issue, I was thinking of https://github.com/evryfs/github-actions-runner-operator/issues/232 which was fixed recently. Still need this to fix this one (deleting CR)

**Comment by davidkarlsen** [comment]
@aroemen #264 will solve this, as you can scale the pool to zero, then delete the CR.

**Comment by zhsj** [comment]
Maybe the CR should have finalizer as well.

**Comment by tonywildey-valstro** [comment]
I'm trying to make this work on latest build but cant seem to make it...  
`$ kubectl patch githubactionrunners.garo.tietoevry.com runner-pool --namespace actions-runner --patch '{"spec":{"minRunners":0}}' --type=merge`
Results in 
`The GithubActionRunner "runner-pool" is invalid: spec.minRunners: Invalid value: 0: spec.minRunners in body should be greater than or equal to 1`

I suspect that either the image i'm pulling is not the latest - or i'm pulling the image wrong, the operator image i'm pulling using the published helm charts  :  
`helm upgrade --install  --wait github-actions-runner-operator evryfs-oss/github-actions-runner-operator --namespace actions-runner-operator --set githubapp.existingSecret=github-runner-app --set githubapp.enabled=true`

The runner image is this one : 
`quay.io/evryfs/github-actions-runner:latest`

What am i missing ? 

Thx
Tony

**Comment by davidkarlsen** [comment]
@tonywildey-valstro you probably don't have the lastest crd: https://raw.githubusercontent.com/evryfs/github-actions-runner-operator/v0.10.0/config/crd/bases/garo.tietoevry.com_githubactionrunners.yaml

**Comment by tonywildey-valstro** [comment]
> @tonywildey-valstro you probably don't have the lastest crd: https://raw.githubusercontent.com/evryfs/github-actions-runner-operator/v0.10.0/config/crd/bases/garo.tietoevry.com_githubactionrunners.yaml

Ah - there we go - I installed using the helm chart: https://github.com/evryfs/helm-charts/blob/master/charts/github-actions-runner-operator/crds/garo.tietoevry.com_githubactionrunners.yaml which does not have the min runners change 

Thx
Tony
